NASA-AMMOS/aerie-ui
# Add Plan Collaborators

Add and remove plan collaborators from the Plan Metadata panel in a plan. Closes #1183.

#### Test Cases:
Setup:
1. Login as userA, userB, and then userC to create those users.

Add Plan Collaborators:
1. Make planA as userA
2. Open the Plan Metadata panel
3. Start typing userA into the input
4. userA should appear as a suggestion
5. Hit enter or click on the suggestion (same behavior as tags component) to add the user as a collaborator

Plan Collaborator Permissions:
1. First run the previous test
2. Login as userB and set your role to "user" instead of "aerie_admin"
3. Go to planA (owned by userA) and ensure that you cannot edit the plan metadata, create/edit activities, etc
4. Switch to "aerie_admin" role and add userB as a plan collaborator
5. Switch back to "user" role and ensure that you can now edit plan collaborators, create/edit activities, etc

Add Plan Collaborators from a Plan:
1. Make planB and add some users as collaborators
2. Make planC and click on the plan collaborator input
3. planB should appear as a suggestion and should have a user icon with a number that equals the number of collaborators + 1 for the owner (if not a collaborator) from planB.
4. Type in "planB" into the input and ensure that planB appears in the suggestions list
5. Click on the planB option
6. All collaborators + the owner should be added as collaborators to planC.
